src/index.ts

```diff
@@ -0,0 +1,61 @@
+interface Field {
+  left: number;
+  top: number;
+  width: number;
+  height: number;
+  chars: string;
+}
+
+export type Fieldset = Field[];
+
+function getNumberFromWords(words: string[]): string {
+  if(!words.length){
+    throw new Error('Words cannot be an empty array');
+  }
+  const number = words.pop();
+  if(number && number.match(/[A-Za-z\d]/) !== null){
+    return number;
+  }
+  return getNumberFromWords(words);
```

Comment:
Keep looping through this function until we find the last word that contains the correct characters